src/server/scripts/Commands/cs_misc.cpp

```diff
@@ -1513,65 +1565,34 @@ class misc_commandscript : public CommandScript
         return true;
     }
 
-    static bool HandleAddItemCommand(ChatHandler* handler, char const* args)
+    static bool HandleAddItemCommand(ChatHandler* handler, ItemTemplate const* itemTemplate, Optional<int32> _count)
     {
-        if (!*args)
+        if (!sObjectMgr->GetItemTemplate(itemTemplate->ItemId))
+        {
+            handler->PSendSysMessage(LANG_COMMAND_ITEMIDINVALID, itemTemplate->ItemId);
+            handler->SetSentErrorMessage(true);
             return false;
+        }
 
         uint32 itemId = 0;
+        int32 count = 1;
 
-        if (args[0] == '[')                                        // [name] manual form
-        {
-            char const* itemNameStr = strtok((char*)args, "]");
-
-            if (itemNameStr && itemNameStr[0])
-            {
-                std::string itemName = itemNameStr + 1;
-
-                WorldDatabasePreparedStatement* stmt = WorldDatabase.GetPreparedStatement(WORLD_SEL_ITEM_TEMPLATE_BY_NAME);
-                stmt->setString(0, itemName);
-                PreparedQueryResult result = WorldDatabase.Query(stmt);
-
-                if (!result)
-                {
-                    handler->PSendSysMessage(LANG_COMMAND_COULDNOTFIND, itemNameStr + 1);
-                    handler->SetSentErrorMessage(true);
-                    return false;
-                }
-                itemId = result->Fetch()->GetUInt32();
-            }
-            else
-                return false;
-        }
-        else                                                    // item_id or [name] Shift-click form |color|Hitem:item_id:0:0:0|h[name]|h|r
+        if (_count)
         {
-            char const* id = handler->extractKeyFromLink((char*)args, "Hitem");
-            if (!id)
-                return false;
-            itemId = uint32(atol(id));
+            count = *_count;
         }
 
-        char const* ccount = strtok(nullptr, " ");
-
-        int32 count = 1;
-
-        if (ccount)
-            count = strtol(ccount, nullptr, 10);
-
-        if (count == 0)
+        if (!count)
+        {
             count = 1;
+        }
```

Comment:
This if is kinda redundant as count is already declared as 1.

Reply:
no, `_count` can = 0

Reply:
Yeah but if _count = 0, then count would be count = 1 anyways, because if _count = 0, then the previous if is not executed